experimental/nomad_resource_plugin/licensedevice/sqldb/sqldb.go

```diff
@@ -24,66 +27,128 @@ const (
 	stateInUse    = "IN_USE"
 )
 
+var (
+	metricGetCurrentDuration = promauto.NewHistogram(prometheus.HistogramOpts{
+		Namespace: "licensedevice",
+		Subsystem: "sqldb",
+		Name:      "get_current_duration_seconds",
+		Help:      "GetCurrent execution time",
+	})
+	metricMyLicenses = promauto.NewGauge(prometheus.GaugeOpts{
+		Namespace: "licensedevice",
+		Subsystem: "sqldb",
+		Name:      "my_licenses",
+		Help:      "How many licenses do I currently have",
+	})
+	metricSqlFailCounter = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "licensedevice",
+		Subsystem: "sqldb",
+		Name:      "sql_failiures",
+		Help:      "The number of times sql has errored in various sections of the code",
```

Comment:
Typically I'll put a counter at a specific point in the code (e.g, counting `Foo()` function executions, by incrementing a counter at `Foo`'s exit) and then count successful executions into an `outcome=ok` label, and error executions into an `outcome=some_err_desc` label.

This allows one to count:
* successful result rate
* error rate (broken down by error)

...and this can be rolled into a histogram as well, if execution time of `Foo` is important.

---

This has the advantage of one counter for all errors, but it doesn't convey what the non-error rate is; instrumenting at a particular place in the code ("all exit points of Foo on a single counter, with different exit points having different label values") gives a more complete picture of whether the code is behaving as expected (allows us to see issues like repeated successful spamming on the DB, for instance)

"various sections of the code" could reduce the power of this counter as well - if events are received, is it clear where in the code the error is coming from? (by inspection of the labels, this appears to be the case - but if a label is reused, it may get confusing)